### ISSUE TITLE
Update RxJS to latest version

### ISSUE THREAD

**Issue by robwormald** [angular/angular#9549]
We need to update RxJS to the newest version (beta8 at the current time)

This needs to be orchestrated as its going to add a dependency that has not previously existed (which will need to be brought into google3 as well) - updates will be required to the quickstart and CLI. 

Tasks:
- [ ] update RxJS version
- [ ] use symbol-observable to do our `isObservable` check in the facade
- [ ] move symbol-observable into google3
- [ ] Docs changes

cc @IgorMinar @wardbell @hansl

**Comment by nosachamos** [comment]
BTW they got beta 9 out with some nice fixes

**Comment by mLaird** [comment]
I'm using rxjs beta.9  Nothing loads beyond the index.html "Loading" page. I get these Console errors:

Error: (SystemJS) patchProperty/desc.set/wrapFn@http://localhost:3000/node_modules/zone.js/dist/zone.js:769:27
	Zone</ZoneDelegate</ZoneDelegate.prototype.invokeTask@http://localhost:3000/node_modules/zone.js/dist/zone.js:356:24
	Zone</Zone</Zone.prototype.runTask@http://localhost:3000/node_modules/zone.js/dist/zone.js:256:29
	ZoneTask/this.invoke@http://localhost:3000/node_modules/zone.js/dist/zone.js:423:29
	
	Error loading http://localhost:3000/symbol-observable as "symbol-observable" from http://localhost:3000/node_modules/rxjs/Observable.js

I'm at a dead end.

**Comment by ericmartinezr** [comment]
You have to map symbol-observable in your systemjs config.

**Comment by mLaird** [comment]
How do I do that?  Where is the documentation?  I am relying on Google to
explain all this complexity.  I don't want to spend time learning systemjs.

On Tue, Jun 28, 2016 at 3:36 PM, Eric Martinez <notifications@github.com>
wrote:

> You have to map symbol-observable in your systemjs config.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/issues/9549#issuecomment-229157843>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADDSb_K2MwaIMOyisdVSGpEz7IxMmSTfks5qQXetgaJpZM4I9NL->
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy

**Comment by ericmartinezr** [comment]
```
map : {
  'symbol-observable' : 'node_modules/symbol-observable/index.js'
}
```
or similar, that's how you map it, I don't know the exact path. I moved out from systemjs too long ago to webpack.

If you have any question about it, you can ask over here https://gitter.im/angular/angular :+1:

**Comment by meffect** [comment]
systemjs.config.js from angular2 guide, added this

    var map = {
       ...
       'symbol-observable' : 'node_modules/symbol-observable'
    }


    var packages = {
         ...
         'symbol-observable': { defaultExtension: 'js', main: 'index.js' }
    };

**Comment by mLaird** [comment]
Eric, thank you for your attention.  Meffect's info works.  You were really
close - considering you don't even use systemjs, I'm impressed. ;-)

On Tue, Jun 28, 2016 at 6:47 PM, Eric Martinez <notifications@github.com>
wrote:

> map : {
>   'symbol-observable' : 'node_modules/symbol-observable/index.js'
> }
>
> or similar, that's how you map it, I don't know the exact path. I moved
> out from systemjs too long ago to webpack.
>
> If you have any question about it, you can ask over here
> https://gitter.im/angular/angular 👍
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/issues/9549#issuecomment-229207237>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADDSbyuLCjnKNtMPi_9h0CyUl5EIXjp6ks5qQaSfgaJpZM4I9NL->
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy

**Comment by mLaird** [comment]
Meffect,  thank you.  This solves the problem.

On Wed, Jun 29, 2016 at 2:40 AM, Meffect <notifications@github.com> wrote:

> systemjs.config.js from angular2 guide, added this
>
> var map = {
>    ...
>    'symbol-observable' : 'node_modules/symbol-observable'
> }
>
>
> var packages = {
>      ...
>      'symbol-observable': { defaultExtension: 'js', main: 'index.js' }
> };
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/issues/9549#issuecomment-229269679>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADDSb4oRcJH_ub0SlveOpOqv1vcp2xAdks5qQhNwgaJpZM4I9NL->
> .
>



-- 



Mike Laird

12 Arbor Creek Drive, Pittsford, New York 14534
mlaird@alum.mit.edu

". . . remembering on both sides that civility is not a sign of weakness,
and sincerity is always subject to proof. Let us never negotiate out of
fear, but let us never fear to negotiate."  - John F. Kennedy

**Comment by Meligy** [comment]
Here's what's working for me for beta9 and beta10, in a project generated by Angular CLI beta9:

1. Add `symbol-observable` to `dependencies` in `packages.config`
2. In `angular-cli-build.js`:  
  In `vendorNpmFiles`, add:  
  `'symbol-observable/**/*.+(js|js.map)',`
3. In `src/system-config.ts`:  
  In `barrels`, add, just before `rx`:  
  `'symbol-observable',`
4. In the same file as 3:  
  under `map`, add to the list:  
  `'symbol-observable': 'vendor/symbol-observable'`

The steps are a little simplified as I copied them from guidance I wrote for a friend.

I have the setup in a repo here: https://github.com/Meligy/ng-cli-latest-sample

P.S.
I remember I also needed the es6-shim typings install, which might be already there (I don't remember).
